src/main/java/org/mitre/synthea/engine/Generator.java

```diff
@@ -334,52 +402,231 @@ private synchronized void writeToConsole(Person person, int index, long time, bo
     }
   }
 
-  private long setDemographics(Person person, Demographics city) {
-    person.attributes.put(Person.CITY, city.city);
-    person.attributes.put(Person.STATE, city.state);
-    
-    String race = city.pickRace(person.random);
-    person.attributes.put(Person.RACE, race);
-    String ethnicity = city.ethnicityFromRace((String)person.attributes.get(Person.RACE), person);
-    person.attributes.put(Person.ETHNICITY, ethnicity);
-    String language = city.languageFromEthnicity((String) person.attributes.get(Person.ETHNICITY),
-        person);
-    person.attributes.put(Person.FIRST_LANGUAGE, language);
-
-    String gender = city.pickGender(person.random);
-    if (gender.equalsIgnoreCase("male") || gender.equalsIgnoreCase("M")) {
-      gender = "M";
+  private long setDemographics(Person person, Demographics city) throws IOException {
+    if (householdsMode.equals("true")) {
+      // get a random spew person
+
+      int[] range = new int[] { 1, spewPerson.size() + 1 };
+
+      int rand = (int) person.rand(range);
+      
+      LinkedHashMap<String, String> randSpew = spewPerson.get(rand);
+
+      String spewSerial = randSpew.get("SERIALNO");
+      person.attributes.put(Person.SPEW_SERIAL_NO, spewSerial);
+
+      String householdIncome = randSpew.get("HINCP");
+      person.attributes.put(Person.HOUSEHOLD_INCOME, householdIncome);
+
+      String householdSize = randSpew.get("NP");
+      person.attributes.put(Person.HOUSEHOLD_SIZE, householdSize);
+
+      // this will have to change based on SPEW latitude/longitude
+      person.attributes.put(Person.CITY, city.city);
+
+      // TODO spew location changes
+      person.attributes.put(Person.STATE, city.state);
+
+      String race = randSpew.get("RAC1P");
+      String hisp = randSpew.get("HISP");
+
+      // race codes, hispanic is a different variable
+      if (race.equals("1")) {
```

Comment:
I had one last thought. There's an odd combination of translations/code mappings used here. For example this is a map from (1..9) to the various races, using an if-else branch. But other places read from CSV files and iterate over the possible values, and certain others use regular lookup tables. It would be a lot cleaner if all of these were moved into individual CSVs and then converted into regular lookup tables. For example here it would be a lot easier if you had a Map<String, String> containing `{ "1" => "white", "2" => "black", "3" => "native", etc... }` and then you just say `person.attributes.put(Person.RACE, map.get(race));` in one line. 

My suggestion for a CSV-to-lookup table is a Utilities function like the following:
```
  /**
   * Convert the parsed CSV into a map of key column => value column.
   * Given a List of Map&lt;String,String&gt;, this method "rotates" it such that
   * a lookup can be done from the key column -> value column.
   * For example, given the following CSV: <pre>
   * CODE,DEFINITION
   * 1,AAA
   * 2,BBB
   * 3,CCC</pre>
   * This function may be called with keyColumn="CODE" and valueColumn="DEFINITION" to produce
   * the following: { "1" => "AAA", "2" => "BBB", "3" => "CCC" }
   * Any other columns in the given map are ignored.
   * @param parsedCsv List of data parsed from a CSV file
   * @param keyColumn Title of the column containing the "key"
   * @param valueColumn Title of the column containing the "value"
   * @return Map of keyColumn => valueColumn
   */
  public static final Map<String, String> createMapFromCsv(
      List<LinkedHashMap<String, String>> parsedCsv, String keyColumn, String valueColumn) {
    Map<String, String> map = new HashMap<>();

    for (Map<String, String> line : parsedCsv) {
      String key = line.get(keyColumn);
      String value = line.get(valueColumn);
      map.put(key, value);
    }

    return map;
 }
```

And a free unit test:
```
  @Test
  public void testCreateMap() throws IOException {
    List<LinkedHashMap<String,String>> parsedCsv =
        SimpleCSV.parse(Utilities.readResource("sampleCsv.csv"));

    Map<String,String> map = Utilities.createMapFromCsv(parsedCsv, "CODE", "DESCRIPTION");

    assertEquals("Examplitis", map.get("123"));
    assertEquals("Examplitol", map.get("456"));
    assertEquals("Examplotomy Encounter", map.get("ABC"));
    assertEquals("Examplotomy", map.get("789"));
 }
```

With a new test resource:  sampleCsv.csv
```
CODE,DESCRIPTION
123,Examplitis
456,Examplitol
ABC,Examplotomy Encounter
789,Examplotomy
```